prdoc/pr_2942.prdoc

```diff
@@ -0,0 +1,13 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Fix pallet-nomination-pools v6 to v7 migration
+
+doc:
+  - audience: Node Dev
+    description: |
+      Restores the behaviour of the nomination pools `V6ToV7` migration so that it still works when
+      the pallet will be upgraded to V8 afterwards.
```

Comment:
This description doesn't make sense to me, V6ToV7 will work regardless of V8 as it is always migrated before V8 right? Or is it fixing something that mod v8 broke?

Reply:
> V6ToV7 will work regardless

No sadly not. It should though - which is what this MR is correcting.  
The migration tries to access storage from `crate::pallet::BondedPoolInner` etc, but this could still be unmigrated since the V8 migration could run afterwards.  

It did work fine in the past, because the pallet storage definitions were not changed to V8 yet.